lib/active_merchant/billing/gateways/paymentez.rb

```diff
@@ -77,13 +77,21 @@ def authorize(money, payment, options = {})
         commit_transaction('authorize', post)
       end
 
-      def capture(money, authorization, _options = {})
+      def capture(money, authorization, options = {})
         post = {
           transaction: { id: authorization }
         }
-        post[:order] = { amount: amount(money).to_f } if money
+        verify_flow = options[:type] && options[:value]
```

Comment:
I think this is good, because it conforms to the fields required by the Paymentez API.

One thing I wonder about is whether a combined field, like `options[:otp_value]`, that sets `post[:type] = 'BY_OTP'` and `post[:value] = options[:otp_value]` would be a benefit for the customer in this use case? It could make things more simple for them to handle on their end, because they only have to worry about passing the OTP as `otp_value` and then we do the rest.

☝️ I thought for a long time about this and I decided that the way you have it right now makes sense. I am just trying to think about things from a customer's perspective and what is easiest for us to maintain in the long run.

Reply:
Yes, I had thought of this approach but the reason I am not in the favor of this approach is that the paymentez API has more `type` & `value` pairs other than just **BY_OTP**. so if we will have this fixed otp_value in the options then we will need to manually set and fix other [4 options for these pairs](https://paymentez.github.io/api-doc/#payment-methods-cards-verify) as well and also in future if new types are added in the API we will manually need to update this in our implementation as well, so what are your suggestions on this?

Reply:
Yes, that is a very good point. I think the way you have it works just fine.

But, if we wanted to preserve the ability for a merchant to pass in a different `type`, we could combine both approaches, like this:

```
verify_flow = verify_flow?(options)

# later in the class
def verify_flow?(options)
  options[:otp_value] || (options[:type] && options[:value])
end

# even later in the class
def add verify_value(post, options)
  type, value = if options[:otp_value]
    ['BY_OTP', options[:otp_value]]
  else
    [options[:type], options[:value]]
  end

  post[:type] = type if type
  post[:value] = value if value
end
```
This way, we could add the convenience `otp_value` for our primary use-case, but if somebody in the future wanted to use this verify flow for a different purpose (say for example passing in a `type` of `'BY_AMOUNT'`), then the adapter already supports that.

I'm not sure if this is better or not. It is nice to have `otp_value` to make it easier to use for the customer, but it also introduces multiple options which could be confusing, and it makes the code a little more complex. So it's a judgement call. What do you think?

Reply:
In my opinion, this may add extra complexity to the code and more importantly this may make our request bit inconsistent with the original parameters being passed in the API and may create some confusion for the customers as it also creates some separation in logic for otp and other type value pairs without some documentation on API. If you suggest to go with this approach then I am also good to go on this as well 😄

Reply:
Ok, I like your reasoning. Let's leave it the way you have it now 👍